File: convertToSrt.cs
using System;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        // binフォルダ内部
        string folderPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "txt");
        ConvertTxtToSrtInFolder(folderPath);
    }

    static List<string> ConvertTxtToSrt(string txtContent)
    {
        string[] lines = txtContent.Split('\n');
        List<string> srtLines = new List<string>();

        for (int i = 0; i < lines.Length; i++)
        {
            string line = lines[i].Trim();
            string timePattern = @"(\d{2}:\d{2}:\d{2}:\d{2}) - (\d{2}:\d{2}:\d{2}:\d{2})";

            if (Regex.IsMatch(line, timePattern))
            {
                // タイムコードの行
                string startTime = Regex.Match(line, timePattern).Groups[1].Value.Replace(":", ",");
                string endTime = Regex.Match(line, timePattern).Groups[2].Value.Replace(":", ",");
                srtLines.Add($"{(i / 2) + 1}");
                srtLines.Add($"{startTime}0 --> {endTime}0");
            }
            else if (!string.IsNullOrWhiteSpace(line))
            {
                // 発話テキストの行
                srtLines.Add(line);
            }
        }

        return srtLines;
    }

    static void ConvertTxtToSrtInFolder(string folderPath)
    {
        string[] txtFiles = Directory.GetFiles(folderPath, "*.txt");

        foreach (string txtFile in txtFiles)
        {
            string txtContent = File.ReadAllText(txtFile, Encoding.UTF8);
            List<string> srtLines = ConvertTxtToSrt(txtContent);

            // .txtを.srtに変更して保存
            string srtFile = Path.ChangeExtension(txtFile, ".srt");
            File.WriteAllLines(srtFile, srtLines, Encoding.UTF8);
        }
    }
}

File: convert_to_srt.py
import os
import re

folder_path = "./txt"

def convert_txt_to_srt(txt_content):
    lines = txt_content.split('\n')
    srt_lines = []

    i = 0
    while i < len(lines):
        line = lines[i].strip()
        time_pattern = r'(\d{2}:\d{2}:\d{2}:\d{2}) - (\d{2}:\d{2}:\d{2}:\d{2})'

        if re.match(time_pattern, line):
            # タイムコードの行
            start_time = re.search(time_pattern, line).group(1).replace(':', ',')
            end_time = re.search(time_pattern, line).group(2).replace(':', ',')
            srt_lines.append(str(i // 2 + 1))
            srt_lines.append(f'{start_time}0 --> {end_time}0')
        elif line.strip():
            # 発話テキストの行
            srt_lines.append(line)

        i += 1

    return srt_lines

def convert_txt_to_srt_in_folder(folder_path):
    txt_files = [f for f in os.listdir(folder_path) if f.endswith('.txt')]

    for txt_file in txt_files:
        with open(os.path.join(folder_path, txt_file), 'r', encoding='utf-8') as file:
            txt_content = file.read()
        
        srt_lines = convert_txt_to_srt(txt_content)

        # .txtを.srtに変更して保存
        srt_file = os.path.splitext(txt_file)[0] + '.srt'
        with open(os.path.join(folder_path, srt_file), 'w', encoding='utf-8') as file:
            file.write('\n'.join(srt_lines))

if __name__ == '__main__':
    # binフォルダ内部
    # folder_path = os.path.join(os.path.dirname(__file__), 'txt')
    convert_txt_to_srt_in_folder(folder_path)
